### PR TITLE
feat(events): added a webhook for events objects

### PR DESCRIPTION
**Pull request by sudoku-lord**
_sobs_ too much formatting..... please let me know if I have copy-paste errors

### CODE REVIEW

File: Lob-API-public.yml [hunk 1]
```diff
@@ -321,6 +321,106 @@ tags:
       | 422           | SPECIAL_CHARACTERS_RESTRICTED      | Cannot use special characters for merge variable names.                                                              |
       | 422           | UNEMBEDDED_FONTS                   | The provided PDF contains non-standard unembedded fonts. See description for details.                                |
 
+  - name: Events
+    description: |
+      When various notable things happen within the Lob architecture, Events will be created. To get these events sent to your server
+      automatically when they occur, you can set up [Webhooks](https://docs.lob.com/#webhooks).
```
Comment (jho44) [inline Lob-API-public.yml:327]:
```suggestion
      automatically when they occur, you can set up [Webhooks](#tag/Webhooks).
```

File: Lob-API-public.yml [hunk 2]
```diff
@@ -1312,6 +1413,18 @@ components:
 security:
   - basicAuth: []
 
+x-webhooks:
+  events:
+    post:
+      summary: Events webhook
```
Comment (jho44) [inline Lob-API-public.yml:1419]:
```suggestion
      summary: Events Webhook
```

**Comment by jho44** [comment]
Additional change: https://github.com/lob/lob-openapi/blob/5d14c8c4339b01e870996190ee9647537d8ea4be/Lob-API-public.yml#L1332

should be 
```      subscription [here](#tag/Events).```

File: resources/events/models/events.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    type: string
+    description: Unique identifier prefixed with `us_ver_`.
+    pattern: "^evt_[a-zA-Z0-9_]+$"
+
+  body:
+    type: object
+    description: >-
+      The body of the associated resource as they were at the time of the event, i.e. the [postcard object](https://docs.lob.com/#postcards_object),
+      [the letter object](https://docs.lob.com/#letters_object), [the check object](https://docs.lob.com/#checks_object), [the address object](https://docs.lob.com/#addresses_object), or [the bank account object](https://docs.lob.com/#bankaccounts_object). For .deleted events,
+      the body matches the response for the corresponding delete endpoint for that resource (e.g. the [postcard delete response](https://lob.com/docs#postcards_delete)).
```
Comment (jho44) [inline resources/events/models/events.yml:22]:
```suggestion
      The body of the associated resource as they were at the time of the event, i.e. the [postcard object](#operation/get_postcard),
      [the letter object](#operation/get_letter), [the check object](#operation/get_check), [the address object](#operation/get_address), or [the bank account object](#operation/get_bank_account). For .deleted events,
      the body matches the response for the corresponding delete endpoint for that resource (e.g. the [postcard delete response](#operation/delete_postcard)).
```

File: resources/events/responses/events.yml [hunk 4]
```diff
@@ -0,0 +1,83 @@
+description: Returns an `event` object to the specified server.
+
+headers:
+  ratelimit-limit:
+    $ref: "../../../shared/headers/ratelimit.yml#/ratelimit-limit"
+  ratelimit-remaining:
+    $ref: "../../../shared/headers/ratelimit.yml#/ratelimit-remaining"
+  ratelimit-reset:
+    $ref: "../../../shared/headers/ratelimit.yml#/ratelimit-reset"
```
Comment (jho44) [inline resources/events/responses/events.yml:9]: Does a webhook count as a POST req and therefore get ratelimit headers? o.o

File: resources/events/models/event_type.yml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+required:
+  - id
+  - enabled_for_test
+  - resource
+  - object
+
+properties:
+  id:
+    oneOf:
```
Comment (jho44) [inline resources/events/models/event_type.yml:11]:
```suggestion
  id:
    description: Unique identifier, full list of possible values can be found [here](#tag/Events).
    oneOf:
```

File: resources/events/models/events.yml [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+  reference_id:
+    type: string
```
Comment (jho44) [inline resources/events/models/events.yml:25]:
```suggestion
  reference_id:
    description: Unique identifier of the related resource for the event.
    type: string
```

File: resources/events/models/events.yml [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+  - body
+  - reference_id
+  - event_type
+  - date_created
+  - object
+
+properties:
+  id:
+    type: string
+    description: Unique identifier prefixed with `us_ver_`.
```
Comment (jho44) [inline resources/events/models/events.yml:14]:
```suggestion
    description: Unique identifier prefixed with `evt_`.
```

File: Lob-API-public.yml [hunk 8]
```diff
@@ -321,6 +321,106 @@ tags:
       | 422           | SPECIAL_CHARACTERS_RESTRICTED      | Cannot use special characters for merge variable names.                                                              |
       | 422           | UNEMBEDDED_FONTS                   | The provided PDF contains non-standard unembedded fonts. See description for details.                                |
 
+  - name: Events
+    description: |
+      When various notable things happen within the Lob architecture, Events will be created. To get these events sent to your server
+      automatically when they occur, you can set up [Webhooks](https://docs.lob.com/#webhooks).
+
+      **<span style="color:#6683a2">Postcards</span>**
```
Comment (jho44) [inline Lob-API-public.yml:329]: If you could replace these colored and bolded spans with <h3>, that'd keep styling with the recent UI stuff I pushed consistent.

File: Lob-API-public.yml [hunk 9]
```diff
@@ -321,6 +321,106 @@ tags:
...
+      | `postcard.in_local_area`          | true      | A postcard receives an "In Local Area" [tracking event](https://docs.lob.com/#tracking_event_object).                                                                                          |
+      | `postcard.processed_for_delivery` | true      | A postcard receives a "Processed for Delivery" [tracking event](https://docs.lob.com/#tracking_event_object)                                                                                   |            
+      | `postcard.re-routed`              | true      | A postcard receives an "Re-Routed" [tracking event](https://docs.lob.com/#tracking_event_object).                                                                                              |
+      | `postcard.returned_to_sender`     | true      | A postcard receives a "Returned to Sender" [tracking event](https://docs.lob.com/#tracking_event_object)                                                                                       |                                                                                                                                  |
```
Comment (jho44) [inline Lob-API-public.yml:342]: Noting these need to be changed when tracking events get merged in. No point in changing them now and will make it easier to find later.

File: Lob-API-public.yml [hunk 10]
```diff
@@ -321,6 +321,106 @@ tags:
...
+      | `self_mailer.rendered_pdf`           | false     | A self mailer's PDF proof is successfully rendered.                                                                                                                                               |
+      | `self_mailer.rendered_thumbnails`    | false     | A self mailer's thumbnails are successfully rendered.                                                                                                                                             |
+      | `self_mailer.deleted`                | false     | A self mailer is successfully canceled.                                                                                                                                                           |
+      | `self_mailer.mailed`                 | true      | A self mailer receives a "Mailed" [tracking event](https://docs.lob.com/#tracking_event_object). Only enabled for certain [Print & Mail Editions](https://dashboard.lob.com/#/settings/editions). |
```
Comment (jho44) [inline Lob-API-public.yml:353]: Noting that this, and others with the keywords "Only enabled for certain", don't have "Only created in the Live Environment" appended.
image

**Review by jho44 — CHANGES_REQUESTED** [review]
Especially good job with the event types...my eyes

File: resources/events/responses/events.yml (hunk 4 above)
Comment (sudoku-lord) [reply resources/events/responses/events.yml:9]: A webhook is a POST request, yeah, so I'd assumed it also has the ratelimit headers––but good catch, I'll confirm that.

File: Lob-API-public.yml (hunk 9 above)
Comment (sudoku-lord) [reply Lob-API-public.yml:342]: yup, I'll change that once I get that PR merged

File: resources/events/models/event_type.yml (hunk 5 above)
Comment (sudoku-lord) [reply resources/events/models/event_type.yml:11]: gonna put the descriptions in the individual files instead

**Review by jho44 — APPROVED** [review]

**Pull request merged**